x/qgb/keeper/keeper_valset.go

```diff
@@ -43,7 +47,12 @@ func (k Keeper) GetLatestValset(ctx sdk.Context) (*types.Valset, error) {
 			return valset, nil
 		}
 	}
-	// should never execute
+	if lastAvailableNonce > 1 {
+		// this means that the latest valset was pruned and we can return the current one as no significant
+		// changes to it happened
+		currentVs, err := k.GetCurrentValset(ctx)
+		return &currentVs, err
+	}
 	panic(errors.Wrap(sdkerrors.ErrNotFound, "couldn't find latest valset"))
```

Comment:
very optional, but this feels like we could instead check `if lastAvailable == 1` panic, then proceed to return the current one.

[question]
if the first nonce always a valset?

Reply:
> if the first nonce always a valset?

yes

Reply:
[c9e926a](https://github.com/celestiaorg/celestia-app/pull/1799/commits/c9e926a06ad6adda74745ce712dff6d46b861241)